mindmeld/active_learning/classifiers.py

```diff
@@ -0,0 +1,342 @@
+import os
+import logging
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from sklearn.model_selection import StratifiedKFold
+
+from .data_loading import DataBucket
+
+from ..components.nlp import NaturalLanguageProcessor
+from ..constants import ACTIVE_LEARNING_RANDOM_SEED
+
+logger = logging.getLogger(__name__)
+
+
+class Classifier(ABC):
+    def __init__(self, app_path, training_level):
+        self.app_path = app_path
+        self.training_level = training_level
+        self.intent2idx, self.idx2intent, self.domain_indices = self._get_mappings()
+
+    def _get_mappings(self):
+        """Get mappings of intents to indices and the indices that map to each domain.
+
+        Returns:
+            intent2idx (Dict): Maps intents to indices
+            idx2intent (Dict): Maps indices to intents
+            domain_indices (Dict): Maps domains to a tuple containing the start and
+                ending indexes of intents with the given domain.
+        """
+        idx = 0
+        intent2idx, idx2intent, domain_indices = {}, {}, {}
+        for domain in sorted(os.listdir(os.path.join(self.app_path, "domains"))):
+            start_idx = idx
+            for intent in sorted(
+                os.listdir(os.path.join(self.app_path, "domains", domain))
+            ):
+                intent2idx[f"{domain}|{intent}"] = idx
+                idx2intent[idx] = f"{domain}|{intent}"
+                idx += 1
+            end_idx = idx - 1
+            domain_indices[domain] = (start_idx, end_idx)
+        return intent2idx, idx2intent, domain_indices
+
+    @abstractmethod
+    def train(self):
+        raise NotImplementedError(
+            "Subclasses must implement their classifier's fit method."
+        )
+
+
+class MindMeldClassifier(Classifier):
```

Comment:
What exactly is a MindMeldClassifier? A description here would be helpful.

Reply:
Added